test/integration/toolchains/gnu/test_pkgconfigdeps.py

```diff
@@ -1118,3 +1171,44 @@ def test_using_deployer_folder():
     assert "libdir=${prefix}/lib" in content
     assert "includedir=${prefix}/include" in content
     assert "bindir=${prefix}/bin" in content
+
+
+def test_pkg_config_deps_set_property():
+    c = TestClient()
+    app = textwrap.dedent("""\
+        import os
+        from conan import ConanFile
+        from conan.tools.gnu import PkgConfigDeps
+        class Pkg(ConanFile):
+            settings = "build_type"
+            requires = "dep/0.1", "other/0.1"
+            def generate(self):
+                pc = PkgConfigDeps(self)
+                pc.set_property("dep", "pkg_config_name", "depx264")
+                pc.set_property("other::mycomp1", "nosoname", True)
+                pc.generate()
+            """)
+
+    pkg_info = {"components": {"mycomp1": {"libs": ["mylib"]}}}
+    c.save({"dep/conanfile.py": GenConanfile("dep", "0.1").with_package_type("shared-library"),
+            "other/conanfile.py": GenConanfile("other", "0.1").with_package_type("shared-library")
+                                                              .with_package_info(pkg_info, {}),
+            "app/conanfile.py": app})
+    c.run("create dep")
+    c.run("create other")
+    c.run("install app")
+    try:
+        # This file shouldn't exist
+        c.load("app/dep.pc")
+        assert False
+    except FileNotFoundError:
+        pass
```

Comment:
We could do it more direct:

```suggestion
    assert not os.path.exists(os.path.join(c.current_folder, "app", "dep.pc"))
```